cclib/method/moments.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Calculation of electric multipole moments based on data parsed by cclib."""
+
+import numpy
+
+from cclib.parser.utils import convertor
+from cclib.method.calculationmethod import Method
+
+
+class Moments(Method):
+    """This class calculates multipole moments and stores them in
+    `results` attribute as a dictionary whose keys denote the used
+    charge population scheme.
+    """
+    def __init__(self, data):
+        self.required_attrs = ('atomcoords', 'atomcharges')
+        self.results = {}
+        
+        super(Moments, self).__init__(data)
+
+    def __str__(self):
+        """Returns a string representation of the object."""
+        return "Multipole moments of %s" % (self.data)
+
+    def __repr__(self):
+        """Returns a representation of the object."""
+        return 'Moments("%s")' % (self.data)
+
+    def _calculate_dipole(self, charges, coords, origin):
+        """Calculate the dipole moment from the given atomic charges
+        and their coordinates with respect to the origin.
+        """
+        coords_au = convertor(coords, 'Angstrom', 'bohr')
+        
+        if self.data.charge == 0:
+            dipole = numpy.dot(charges, coords_au)
+        else:
+            origin_au = convertor(origin, 'Angstrom', 'bohr')
+            dipole = numpy.dot(charges, coords_au - origin_au)
+            
+        return convertor(dipole, 'ebohr', 'Debye')
+
+    def _calculate_quadrupole(self, charges, coords, origin):
+        """Calculate the traceless quadrupole moment from the given
+        atomic charges and their coordinates with respect to the origin.
+        """
+        transl_coords_au = convertor(coords - origin, 'Angstrom', 'bohr')
+
+        delta = numpy.eye(3)
+        Q = numpy.zeros([3, 3])
+        for i in range(3):
+            for j in range(3):
+                for q, r in zip(charges, transl_coords_au):
+                    Q[i,j] += 1/2 * q * (3 * r[i] * r[j] - \
```

Comment:
Please notice that I changed the way to calculate the (traceless) quadrupole moment tensor Q to make it consistent with the Buckingham definition. There is another definition (it differs by multiplication by 2), e.g., given on p. 99 of section 41 in the reference below:

L.D. Landau and E.M. Lifshitz, The Classical Theory of Fields (4th ed.), Pergamon Press, Oxford (1994).

Reply:
Could you state that as a short comment in the code?

Also, would you be able to find the DOI for the Buckingham reference and add it?

Reply:
It would be ideal to have both versions.

Reply:
@berquist Done it. I added another reference mentioned by Buckingham, 1959 in his article.

@langner There's two options from me: one, add optional argument, say, `quad` with 'buckingham' and 'hirschfelder' acceptable values. Or two, leave it as is. After reading docstring the end-user can decide for himself—to multiply or not to multiply.

Reply:
That's fine.